src/components/manage-registry/manage-agents/ManageAgents.tsx

```diff
@@ -67,10 +84,53 @@ export function ManageAgents({
     },
   });
 
+  const addNewUserAgentAsAdmin = async () => {
+    if (filteredAgents.includes(selectedAgentToAdd)) {
+      setAgentSelectError("Agent already exists");
+
+      return;
+    } else {
+      setAgentSelectError(undefined);
+    }
+
+    if (selectedAgentToAdd && ethers.isAddress(selectedAgentToAdd)) {
+      try {
+        const txHash = await mutateIdentityRegistryAddAgent(selectedAgentToAdd);
+        setUpdateTxResult(txHash);
+        setUpdateTxError(undefined);
+        setTimeout(() => {
```

Comment:
I do not consider random timeouts to be a reliable way to handle things, let's try to avoid them.

If the idea is to clear cache after successful mutation, there are things like onSuccess etc available
https://tanstack.com/query/latest/docs/framework/react/guides/mutations#mutation-side-effects